common/src/main/java/io/homeassistant/companion/android/common/data/keychain/KeyChainRepositoryImpl.kt

```diff
@@ -32,15 +29,8 @@ class KeyChainRepositoryImpl @Inject constructor(
             alias = prefsRepository.getKeyAlias()
         }
 
-        if (alias != null && !isLoading && (key == null || chain == null)) {
-            isLoading = true
-            if (chain == null) {
-                chain = KeyChain.getCertificateChain(context, alias!!)
-            }
-            if (key == null) {
-                key = KeyChain.getPrivateKey(context, alias!!)
-            }
-            isLoading = false
+        runBlocking {
+            doLoad(context)
         }
```

Comment:
There's no need to wrap it in `runBlocking` because `doLoad` is not a `suspend`ing function. See the [documentation of `runBlocking`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/run-blocking.html)

```suggestion
        doLoad(context)
```